app/src/main/java/app/grapheneos/camera/capturer/ImageSaver.kt

```diff
@@ -73,17 +74,33 @@ class ImageSaver(
     val contentResolver = appContext.contentResolver
     val mainThreadExecutor = appContext.mainExecutor
 
+    private var isCancelled = false
+
+    fun cancelCaptureRequest() {
+        isCancelled = true
+    }
+
     override fun onCaptureSuccess(image: ImageProxy) {
-        mainThreadExecutor.execute(imageCapturer::onCaptureSuccess)
+        mainThreadExecutor.execute mainExecutor@ {
+            if (isCancelled) return@mainExecutor
 
-        try {
-            extractJpegBytes(image)
-        } catch (e: Exception) {
-            handleError(ImageSaverException(Place.IMAGE_EXTRACTION, e))
-            return
+            imageCapturer.onCaptureSuccess()
+
+            imageCaptureCallbackExecutor.execute iccExecutor@ {
```

Comment:
Why is this call delayed until after `imageCapturer.onCaptureSuccess()`?

Reply:
We could re-order the calls here. I'll make the changes for the same soon.